### PR TITLE
incremental learning

### PR DESCRIPTION
**Pull request by CharlesAuthier**

### CODE REVIEW

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+For a total of three losses:
+- $$L_{class}$$ (sigmoid cross entropy loss)
+
+![](/deep-learning/images/incrementallearning/loss1.png)
+
+- $$L_{distill}$$, measures the disparity between $$y_{mem}$$ samples from the memory network, and $$y_{up_prev}$$ from the updated network.
+
+![](/deep-learning/images/incrementallearning/loss2.png)
+
+- $$L_{adapt}$$ is the Loss use during the adaptation
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: is the Loss use during the adaptation ->  is the loss used during the adaptation

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+![](/deep-learning/images/incrementallearning/loss12.png)
+
+- $$L_{rem}$$ is the costum loss for the Incremental learning
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: is the costum loss for the Incremental learning -> is the custom loss for the incremental learning

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+![](/deep-learning/images/incrementallearning/loss3.png)
+
+## Network
+
+This Network is a variant of U-net.
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: This Network -> This network

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+![](/deep-learning/images/incrementallearning/network.png)
+
+- Static learning: This is the traditional learning approach, where they assume that all the training images and annotations for the same classes are available at the time of training.
+
+- Multiple learning: In this learning strategy, they train an additional classifier whenever the new training data are obtained. The number of classifiers that needs to be stored increases linearly.
+
+- Fixed representation: To learn new classes, they remove the classification layers, which were optimized for the previous classes and plug in new classification layers dedicated for the new classes. The newly added classification layers are initialized with Xavier method[^footnote]. When new training data arrive, thay optimize only the newly added classification layers and freeze the rest of the network.
+
+- Fine-tuning: They use a similar strategy that fixed representation. The only difference is that while training the network, instead of only the classification layers, they optimize the whole network using only the new training data.
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: They use a similar strategy that fixed representation. -> They use a similar strategy to fixed representation. (?)

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+
+- Incremental learning: This strategy required for the memory network to generate probability maps from the training patches to optimize $$L_distil$$.
```
Comment (jhlegarreta) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: This strategy required (...) -> This strategy require (...) 
for the sake of consistence in tense (?)

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 6]
```diff
@@ -0,0 +1,76 @@
+---
+layout: review
+title:  "Incremental Learning for Semantic Segmentation of Large-Scale Remote Sensing Data"
+tags:   Incremental learning, catastrophic forgetting, deep-learning, segmentation, remote-sensing,
```
Comment (jodoin) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: only put tags that are in the howto page : https://vitalab.github.io/howto/

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  venue:   "Arxiv, 29 oct 2018"
+---
+
+Code: https://project.inria.fr/epitome/inc_learn/
+
+## Method
+One big probleme, catastrophic forgetting: a significant performance drop for the already learned classes when new classes are added on the data, having no annotations for the old classes.
+They propose an incremental learning methodology, enabling to learn segmenting new classes without hindering dense labeling abilities for the previous classes, although the entire previous data are not accessible.
+
+The key points of the proposed approach are adapting the network to learn new as well as old classes on the new training data, and allowing it to remember the previously learned information for the old classes.
```
Comment (jodoin) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: The proposed approach adapts the network to learn....
and allows it to remember

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+![](/deep-learning/images/incrementallearning/loss3.png)
+
+## Network
+
+This Network is a variant of U-net.
+
+![](/deep-learning/images/incrementallearning/network.png)
+
+- Static learning: This is the traditional learning approach, where they assume that all the training images and annotations for the same classes are available at the time of training.
```
Comment (jodoin) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: learning approach which assumes that 100% of the training images...

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 9]
```diff
@@ -0,0 +1,76 @@
...
+
+- Multiple learning: In this learning strategy, they train an additional classifier whenever the new training data are obtained. The number of classifiers that needs to be stored increases linearly.
```
Comment (jodoin) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: this learning strategy trains an ...
the new training data are obtained.-> new training data is made available.

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 10]
```diff
@@ -0,0 +1,76 @@
...
+
+- Fixed representation: To learn new classes, they remove the classification layers, which were optimized for the previous classes and plug in new classification layers dedicated for the new classes. The newly added classification layers are initialized with Xavier method[^footnote]. When new training data arrive, thay optimize only the newly added classification layers and freeze the rest of the network.
```
Comment (jodoin) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: When new training data is made available, only the newly added classification layers are optimized, the rest of the network being frozen.

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+
+- Fine-tuning: They use a similar strategy that fixed representation. The only difference is that while training the network, instead of only the classification layers, they optimize the whole network using only the new training data.
+
+- Incremental learning: This strategy required for the memory network to generate probability maps from the training patches to optimize $$L_distil$$.
+
+![](/deep-learning/images/incrementallearning/compare.png)
+
+## Results
+
+Not the new state-of-the-art, but show that it is not far and allow the network to learn new classes.
```
Comment (jodoin) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: Not better than state-of-the-art, ...
allows the network

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md (hunk 2 above)
Comment (jhlegarreta) [reply deep-learning/_posts/2018-11-09-Incremental-Learning.md]: Has not been yet changed. Am I misunderstanding the meaning of the term?

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md (hunk 5 above)
Comment (jhlegarreta) [reply deep-learning/_posts/2018-11-09-Incremental-Learning.md]: **Sorry** I must have said "This strategy requires (...)"

**Comment by jhlegarreta** [comment]
Looks like a change to the file `Gemfile.lock` was included in commit 20a3b95. Was this intended? If a new version of the file is required, I think it'd be cleaner to push it in another PR, since in principle it is unrelated to a review, and stating why this change is necessary. Thanks.

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+
+- $$L_{distill}$$, measures the disparity between $$y_{mem}$$ samples from the memory network, and $$y_{up_prev}$$ from the updated network.
+
+![](/deep-learning/images/incrementallearning/loss2.png)
+
+- $$L_{adapt}$$  is the loss used during the adaptation
+
+![](/deep-learning/images/incrementallearning/loss12.png)
+
+- $$L_{rem}$$ is the costum loss for the incremental learning
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: custom

File: deep-learning/_posts/2018-11-09-Incremental-Learning.md [hunk 13]
```diff
@@ -0,0 +1,76 @@
...
+
+- Fine-tuning: They use a similar strategy to fixed representation. The only difference is that while training the network, instead of only the classification layers, they optimize the whole network using only the new training data.
+
+- Incremental learning: This strategy requires for the memory network to generate probability maps from the training patches to optimize $$L_distil$$.
+
+![](/deep-learning/images/incrementallearning/compare.png)
+
+## Results
+
+Not better than state-of-the-art, but show that it is not far and allows the network to learn new classes.
```
Comment (lemairecarl) [inline deep-learning/_posts/2018-11-09-Incremental-Learning.md]: I would underline the fact that the goal of this paper is not to beat single dataset methods.

-> "The results are competitive with the state-of-the-art single-dataset methods, while allowing the network to learn new classes."

Is this correct? Maybe I'm mistaken.
